MetaMask/metamask-extension
# Sign in with Ethereum: re-enable warning UI for mismatched domains / disable domain binding

## Explanation

Fixes https://github.com/MetaMask/metamask-extension/issues/17707

Hotfix targeting v10.26.2

- unblocks mismatched domain support
- reverts logic from #16616
- we may re-add error handling here with proper error handling #18184
- temp fix SIWE actionable-message styles
  - Improved style fix in follow-up PR since it may effect other <ActionableMessage> instances: https://github.com/MetaMask/metamask-extension/pull/18207



## Screenshots/Screencaps



### Before

https://user-images.githubusercontent.com/53189696/218042372-ef68235e-51a1-402d-a696-ec8716c5d7f5.mov

previous warning UI:
<img width="406" alt="Screen Shot 2023-03-16 at 10 22 22 AM" src="https://user-images.githubusercontent.com/20778143/225701852-2ff80d22-227a-4f31-83f4-ef30696c6083.png">



### After
<img width="392" alt="Screen Shot 2023-03-16 at 10 22 29 AM" src="https://user-images.githubusercontent.com/20778143/225701862-c35e77a6-c712-457d-ab1e-8955817eaccb.png">
<img width="378" alt="Screen Shot 2023-03-16 at 10 22 13 AM" src="https://user-images.githubusercontent.com/20778143/225701844-ace4ec55-2886-43f8-85dd-5c4fbfc81468.png">
<img width="386" alt="Screen Shot 2023-03-16 at 10 38 04 AM" src="https://user-images.githubusercontent.com/20778143/225710410-d631bb68-32dc-487e-9369-490d448725cd.png">

## Manual Testing Steps

1. Login to the extension
2. Navigate to the test dapp
3. Click Sign In With Ethereum (Bad Account)
4. Observe warning flow 



## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
